cmake/caches/LLVM.cmake

```diff
@@ -0,0 +1,98 @@
+    set(LLVM_ENABLE_EH ON CACHE BOOL "")
+    set(LLVM_ENABLE_RTTI ON CACHE BOOL "")
+    set(LIBCXX_ENABLE_EXCEPTIONS ON CACHE BOOL "")
+    set(LIBCXXABI_ENABLE_EXCEPTIONS ON CACHE BOOL "")
+    set(LIBCXXABI_ENABLE_FORGIVING_DYNAMIC_CAST ON CACHE BOOL "")
+
+    # Path configurations
+    set(LLVM_ENABLE_PER_TARGET_RUNTIME_DIR ON CACHE BOOL "")
+    set(CMAKE_INSTALL_RPATH_USE_LINK_PATH OFF CACHE BOOL "")
+    set(CMAKE_INSTALL_RPATH "$ORIGIN:$ORIGIN/../lib:$ORIGIN/${LLVM_DEFAULT_TARGET_TRIPLE}:$ORIGIN/lib/${LLVM_DEFAULT_TARGET_TRIPLE}:$ORIGIN/../lib/${LLVM_DEFAULT_TARGET_TRIPLE}" CACHE STRING "")
+
+    # Default configurations for the built toolchain
+    set(CLANG_DEFAULT_OPENMP_RUNTIME libomp CACHE STRING "")
+    set(CLANG_DEFAULT_LINKER lld CACHE STRING "")
+    set(CLANG_DEFAULT_CXX_STDLIB libc++ CACHE STRING "")
+    set(CLANG_DEFAULT_RTLIB compiler-rt CACHE STRING "")
+    set(CLANG_DEFAULT_UNWINDLIB libunwind CACHE STRING "")
+
+    # Runtime related build configurations
+    set(LLVM_ENABLE_LIBCXX ON CACHE BOOL "")
+    set(LIBCXX_CXX_ABI libcxxabi CACHE STRING "")
+    set(COMPILER_RT_USE_LIBCXX ON CACHE BOOL "")
+    set(COMPILER_RT_USE_LLVM_UNWINDER ON CACHE BOOL "")
+    set(LIBUNWIND_USE_COMPILER_RT ON CACHE BOOL "")
+    set(LIBCXX_USE_COMPILER_RT ON CACHE BOOL "")
+    set(LIBCXXABI_USE_COMPILER_RT ON CACHE BOOL "")
+    set(LIBCXXABI_USE_LLVM_UNWINDER ON CACHE BOOL "")
+    set(LIBUNWIND_HAS_GCC_LIB OFF CACHE BOOL "")
+    set(LIBUNWIND_HAS_GCC_S_LIB OFF CACHE BOOL "")
+    set(LIBCXXABI_HAS_GCC_LIB OFF CACHE BOOL "")
+    set(LIBCXXABI_HAS_GCC_S_LIB OFF CACHE BOOL "")
+    set(LIBCXX_HAS_GCC_LIB OFF CACHE BOOL "")
+    set(LIBCXX_HAS_GCC_S_LIB OFF CACHE BOOL "")
+    set(COMPILER_RT_HAS_GCC_LIB OFF CACHE BOOL "")
+    set(COMPILER_RT_HAS_GCC_S_LIB OFF CACHE BOOL "")
+    set(COMPILER_RT_BUILD_BUILTINS ON CACHE BOOL "")
+    set(COMPILER_RT_BUILD_LIBFUZZER OFF CACHE BOOL "")
+    set(COMPILER_RT_BUILD_SANITIZERS OFF CACHE BOOL "")
+    set(COMPILER_RT_BUILD_PROFILE OFF CACHE BOOL "")
+    set(COMPILER_RT_BUILD_CTX_PROFILE OFF CACHE BOOL "")
+    set(COMPILER_RT_BUILD_MEMPROF OFF CACHE BOOL "")
+    set(COMPILER_RT_BUILD_XRAY OFF CACHE BOOL "")
+    set(COMPILER_RT_USE_BUILTINS_LIBRARY ON CACHE BOOL "")
+    # This looks like a configuration that was originally used and
+    # probably automatically set to the correct value by the larger
+    # LLVM build infrastructure. However, at the time of writing, 
+    # this setting was only ever used in one place in the LLVM
+    # codebase and never set. Specifically, it is used to determine 
+    # whether to initialize the variables that capture what 
+    # architectures compiler-rt can target.
+    # If this is not set, the check is not performed before 
+    # COMPILER_RT_HAS_CRT is set and the dependent option 
+    # COMPILER_RT_BUILD_CRT has no effect, causing the build to 
+    # not create clang_rt.crtbegin.o and clang_rt.crtend.o.
```

Comment:
This comment is a bit worrisome. It implies we're depending on build variables that aren't well established and likely not well tested, if at all.

Reply:
I am all ears for suggestion how to ensure crtbegin/end are built without setting this. This seems to be an oddity or maybe bug in the llvm build system; the build produces these perfectly fine without explicitly setting the variable in some cases (I can't remember exactly which setting ultimately made this necessary, initially - I think without openmp in the mix - I had this building without this)

Reply:
I should also clarify that this variable literally has *only* one effect: to make sure that all viable target architectures are properly listed, which is exactly what we need. It is not used in any other way whatsoever: https://github.com/search?q=repo%3Allvm%2Fllvm-project%20COMPILER_RT_CRT_STANDALONE_BUILD&type=code
So we are not so much relying on a potentially untested configuration here, as we are actually manually forcing it to do the check it needs to do before building - I consider it a oversight in the build that this is gated by this variable.